cpp/src/arrow/util/decimal_test.cc

```diff
@@ -40,6 +40,9 @@ namespace arrow {
 using internal::int128_t;
 using internal::uint128_t;
 
+constexpr int128_t kInt128Max = (static_cast<int128_t>(INT64_MAX) << 64) - 1 +
+                                2 * (static_cast<int128_t>(INT64_MAX) + 1);
```

Comment:
Why not `(static_cast<int128_t>(INT64_MAX) << 64) + static_cast<int128_t>(UINT64_MAX)` ?

Reply:
Changed accordingly. Thx!